lib/osut/utils.rb

```diff
@@ -3334,56 +3396,88 @@ def pointWithinPolygon?(p0 = nil, s = [], entirely = false)
   # @return [Bool] whether 2 polygons are parallel
   # @return [false] if invalid input (see logs)
   def parallel?(p1 = nil, p2 = nil)
-    p1  = poly(p1, false, true, false)
-    p2  = poly(p2, false, true, false)
+    p1 = poly(p1, false, true)
+    p2 = poly(p2, false, true)
     return false if p1.empty?
     return false if p2.empty?
 
-    p1 = getNonCollinears(p1, 3)
-    p2 = getNonCollinears(p2, 3)
-    return false if p1.empty?
-    return false if p2.empty?
+    n1 = OpenStudio.getOutwardNormal(p1)
+    n2 = OpenStudio.getOutwardNormal(p2)
+    return false if n1.empty?
+    return false if n2.empty?
+
+    n1.get.dot(n2.get).abs > 0.99
+  end
+
+  ##
+  # Validates whether a polygon can be considered a valid 'roof' surface, as per
+  # ASHRAE 90.1 & Canadian NECBs, i.e. outward normal within 60° from vertical
+  #
+  # @param pts [Set<OpenStudio::Point3d>] OpenStudio 3D points
+  #
+  # @return [Bool] if considered a roof surface
+  # @return [false] if invalid input (see logs)
+  def roof?(pts = nil)
```

Comment:
Single-span, prefab arched structures often end up with tilted side _walls_ that share the same construction as their neighbouring _roof_ segments. Quite common in single story industrial facilities (e.g. [NorsemanStructures](https://www.norsemanstructures.com)):

<img width="536" alt="norseman" src="https://github.com/user-attachments/assets/c1a5ab31-c0f6-4e7d-9a7a-3cd4497197be">

Am sometimes encountering _OpenStudio_ models where such side walls end up typed as **roofs**, possibly to facilitate default construction inheritance. Understandable, yet energy codes (90.1, NECBs) are pretty clear: once a **roof**'s outward normal is tilted too far away from zenith, it's considered a **wall** (i.e. not a candidate surface for skylights).